File: src/com/magayaga/microscript/ListVariable.java
/**
 * MicroScript — The programming language
 * Copyright (c) 2024-2026 Cyril John Magayaga
 * 
 * It was originally written in Java programming language.
 */
package com.magayaga.microscript;

import java.util.ArrayList;

public class ListVariable extends ArrayList<Object> {
    public ListVariable() {
        super();
    }

    public ListVariable(String[] elements) {
        for (String element : elements) {
            this.add(element.trim());
        }
    }
    
    public ListVariable(Object[] elements) {
        for (Object element : elements) {
            this.add(element);
        }
    }
    
    public Object get(int index) {
        return super.get(index);
    }
}

File: src/microscript/ListVariable.cs
/**
 * MicroScript — The programming language
 * Copyright (c) 2024-2025 Cyril John Magayaga
 * 
 * It was originally written in C# programming language.
 */
using System.Collections.Generic;

namespace com.magayaga.microscript
{
    public class ListVariable : List<object>
    {
        public ListVariable() : base() { }

        public ListVariable(string[] elements)
        {
            foreach (var element in elements)
            {
                this.Add(element.Trim());
            }
        }
    }
}